project_task_code/models/project_task.py

```diff
@@ -28,3 +29,46 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
```

Comment:
to avoid having to write the super call twice, better do

```python
tasks = []
...
if code:
    tasks = our_search.name_get()
    if limit and len(tasks) == limit:
        return tasks
return tasks + super()
```